internal/controller/suite_test.go

```diff
@@ -53,10 +57,39 @@ var _ = BeforeSuite(func() {
 	utilruntime.Must(kumquatv1beta1.AddToScheme(scheme))
 
 	// Bootstrap the test environment
+	// Assumes that the manifests, generate, and envtest Makefile targets are up-to-date
 	By("bootstrapping test environment")
+
+	// Open the Makefile and parse it to extract the value of ENVTEST_K8S_VERSION
+	binaryDir := os.Getenv("KUBEBUILDER_ASSETS")
+
+	if binaryDir == "" {
+		makefilePath := filepath.Join("..", "..", "Makefile")
+		makefile, err := os.Open(makefilePath)
+		Expect(err).NotTo(HaveOccurred(), "Failed to open Makefile")
+		defer makefile.Close() // nolint:errcheck
+		scanner := bufio.NewScanner(makefile)
+
+		envtestK8sVersion := "1.30.0" // Default version if ENVTEST_K8S_VERSION is not set
```

Comment:
Do we need to change it in the future if they upgrade the k8s version?

Reply:
I think we will update the `Makefile` with the new version when we want to use a new Kubernetes version.

Then the code after this line will read the new version from the `Makefile` and use that instead.